### DISCUSSION TITLE
Sometimes maps don't show update button

### DISCUSSION THREAD

**Discussion by Feodor0090** [ppy/osu#21174] [Q&A] [answered]
I noticed this a lot of times, but now caught an example.
Last time when i played the game was about 11 hours ago.
[The map](https://osu.ppy.sh/beatmapsets/1878416#osu/3871927) says in the overlay that it was updated 4 hours ago:
![изображение](https://user-images.githubusercontent.com/53872073/200523075-6e2b5363-da86-4e8d-87d8-ce3e67e59b61.png)
But in song select there is no button:
![изображение](https://user-images.githubusercontent.com/53872073/200523798-82cc5996-38aa-4bff-977c-925045f2a05b.png)
Doing anything in song select or entering gameplay does not fix this. But exiting to main menu and going back does:
![изображение](https://user-images.githubusercontent.com/53872073/200524429-56041ae3-a866-4a0c-9c20-3a0209b74253.png)
When this feature was announced it was shown that update button appears after several seconds and stays here, so i guess it's a bug?
[database.log](https://github.com/ppy/osu/files/9959564/database.log)
[network.log](https://github.com/ppy/osu/files/9959565/network.log)
[runtime.log](https://github.com/ppy/osu/files/9959566/runtime.log)

Also, this time this is fixable by exiting to main menu, but yesterday i had a map that showed its update button only after game restart. What information and files should i provide if i notice this again?

**Comment by peppy** [comment, marked as answer]
The issue here seems to be that while the update process does invalidate the `WorkingBeatmap`, the carousel panels are not currently made to respond to these changes.

**Reply by peppy** [reply]
Tracking in https://github.com/ppy/osu/issues/21357.

**Reply by Feodor0090** [reply]
Bumping as i found a case when update button doesn't show even after game restart.
Map in question:
![изображение](https://user-images.githubusercontent.com/53872073/206900308-579530a2-6bf3-4e81-9c85-7e242bf4258f.png)
![изображение](https://user-images.githubusercontent.com/53872073/206900316-bf69edc5-5c23-456a-a1ee-db3620ceb610.png)
https://osu.ppy.sh/beatmapsets/1897551
Logs from first launch today when i updated most maps excluding this one: [logs1.zip](https://github.com/ppy/osu/files/10202008/logs1.zip)
Second: [logs2.zip](https://github.com/ppy/osu/files/10202010/logs2.zip)
Realm jic: [client.realm.zip](https://github.com/ppy/osu/files/10202014/client.realm.zip)

I saw an issue about updating if a new difficulty was added but no exisiting were modified, maybe that's it, not sure
